src/utils/__tests__/async-test.js

```diff
@@ -1,14 +1,47 @@
 /* @flow strict-local */
 import { sleep, tryUntilSuccessful } from '../async';
+import { Lolex } from '../../__tests__/aux/lolex';
 
 describe('sleep', () => {
+  let lolex: Lolex;
+
+  beforeAll(() => {
+    lolex = new Lolex();
+  });
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
   test('waits for a given time in milliseconds', async () => {
     const expectedMs = 1000;
-    const start = Date.now();
-    await sleep(expectedMs);
-    const durationMs = Date.now() - start;
-    expect(expectedMs).toBeLessThanOrEqual(durationMs);
-    expect(durationMs).toBeLessThan(10 * expectedMs);
+
+    const expectationPromise = async () => {
+      const start = Date.now();
+      await sleep(expectedMs);
+      const durationMs = Date.now() - start;
+
+      expect(expectedMs).toBeLessThanOrEqual(durationMs);
+      // cannot be guaranteed: arbitrary delays are possible
+      // expect(durationMs).toBeLessThan(10 * expectedMs);
+
+      return true;
+    };
+    const expectation: Promise<boolean> = expectationPromise();
+
+    // `expectation` should have set up its timer before returning control.
+    expect(lolex.getTimerCount()).toBe(1);
+
+    // (This would cause the commented test in `expectationPromise` to fail.)
+    lolex.advanceTimersByTime(expectedMs * 100);
+
+    // If this stalls, Jest will report failure.
+    expect(await expectation).toBeTrue();
```

Comment:
Hmm interesting! What does this mean -- how does Jest decide that it's stalled, vs. that it just hasn't completed yet (but later will)?

Reply:
There's a configurable timeout, defaulting to 5000ms. [[1]](https://jestjs.io/docs/en/configuration.html#testtimeout-number) [[2]](https://stackoverflow.com/q/49603939)

Reply:
Cool, thanks.

Just "stalls" doesn't communicate that to me. How about "If this never completes, Jest will time out and report failure."?

Reply:
(Expanded somewhat more than even that, including a documentation link.)